doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,289 @@
+.. _mcumgr_smp_protocol_specification:
+
+SMP Protocol Specification
+##########################
+
+This is description of Simple Management Protocol, SMP, that is used by
+mcumgr to pass requests to devices and receive responses from them.
+
+The protocol is an application layer protocol,
+so this documentation explains the protocol independently from a transport
+layer, which may be UART (serial), Bluetooth, UDP or probably other.
+
+
+Frame: The biggest envelope
+***************************
+
+The frame is the base of each SMP request, SMP does not fragment frames although
+transport may do so, but is then also responsible for putting a frame together
+from fragments.
+Each frame consists of header and following it data.
+Frame is encoded in "Big Endian" (Network endianness), where field is more than
```

Comment:
Will the little endian option be dropped? https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/lib/mgmt/include/mgmt/mgmt.h#L67

Reply:
It seems that code has never supported the little-endian version, except for the header definition.